blocks/aside/aside.js

```diff
@@ -0,0 +1,124 @@
+import { createEl } from '../../scripts/scripts.js';
+import { decorateIcons, getMetadata, loadScript } from '../../scripts/lib-franklin.js';
+
+async function generatePDF(pageTitle) {
+  // Source HTMLElement or a string containing HTML.
+  const main = document.querySelector('main').cloneNode(true);
+  const asideContainer = main.querySelector('.aside-container');
+  asideContainer.remove();
+  const pageName = pageTitle.replace(/[^a-z0-9]/gi, '-');
+  const { jsPDF } = window.jspdf;
+  // eslint-disable-next-line new-cap
+  const doc = new jsPDF();
+  await doc.html(main, {
+    // eslint-disable-next-line no-shadow
+    callback(doc) {
+      // Save the PDF
+      doc.save(`${pageName}`);
+    },
+    margin: [10, 10, 10, 10],
+    autoPaging: 'text',
+    x: 0,
+    y: 0,
+    width: 190, // target width in the PDF document
+    windowWidth: 900, // window width in CSS pixels
+  });
+}
+
+export default async function decorate(block) {
+  block.innerText = '';
+  const pageUrl = window.location.href;
+  const pageTitle = getMetadata('title');
```

Comment:
@amol-anand  this is coming as empty.. I think we've meta tag with og:title, but not with title. This is causing the pdf name as `generated.pdf` instead of title.

Reply:
yes! updated.